http/httputils.c

```diff
@@ -0,0 +1,564 @@
+/* SPDX-FileCopyrightText: 2019-2025 Greenbone AG
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+/**
+ * @file httputils.c
+ * @brief HTTP utility functions built on libcurl.
+ *
+ * This module provides an abstraction layer over libcurl to simplify HTTP(S)
+ * request handling. It supports:
+ *
+ * - Synchronous and asynchronous requests (via easy and multi handles).
+ * - Custom HTTP methods, headers, and payloads.
+ * - SSL/TLS configuration (CA certificates, client certs, private keys).
+ * - Response buffering through a write callback.
+ * - Encapsulation of libcurl handles in domain-specific types (e.g., gvm_http_t).
+ */
+
+#include "httputils.h"
+
+#undef G_LOG_DOMAIN
+/**
+ * @brief GLib logging domain.
+ */
+#define G_LOG_DOMAIN "libgvm util"
+
+/**
+ * @brief Allocate gvm http multi handler
+ *
+ * @return gvm http multi handler.
+ */
+static gvm_http_multi_t *
+gvm_http_multi_new (void)
+{
+  return (gvm_http_multi_t *) g_malloc0 (sizeof (struct gvm_http_multi));
+}
+
+/**
+ * @brief Callback function to store the response.
+ * This function is used with libcurl to dynamically store response data.
+ */
+static size_t
+curlutils_response_callback (void *ptr, size_t size, size_t nmemb, void *userdata)
```

Comment:
```suggestion
store_response_data (void *ptr, size_t size, size_t nmemb, void *userdata)
```
or something like that.